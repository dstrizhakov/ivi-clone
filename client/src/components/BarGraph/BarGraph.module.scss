@import "src/styles/variables";

.bar {
  min-width: 10px;
  width: 100%;
  background-color: rgba(126,121,143,.32);
  height: 100%;
  min-height: 3px;
  border-radius: 2px;
  margin: 1px;
  &Value {
    background-color: $color-accent;
    min-height: 3px;
    border-radius: 2px;
  }
}
