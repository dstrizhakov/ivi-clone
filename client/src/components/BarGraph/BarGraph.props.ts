export interface BarGraphProps {
  width: number;
}
