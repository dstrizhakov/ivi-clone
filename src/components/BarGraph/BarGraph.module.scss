@import '@/styles/variables.scss';

.bar {
  min-width: 10px;
  width: 100%;
  background-color: rgba(126,121,143,.32);
  min-height: 3px;
  //border-radius: 2px; //? Если нужно сделать как вариант настраиваемый через пропсы
  margin: 1px;
  &Value {
    min-width: 0 !important;
    max-width: 100% !important;
    background-color: $color-accent;
    min-height: 3px;
    border-bottom-right-radius: 2px;
    border-top-right-radius: 2px;
    //border-radius: 2px ; //? Если нужно сделать как вариант настраиваемый через пропсы
    transition: width 0.5s;
  }
}
