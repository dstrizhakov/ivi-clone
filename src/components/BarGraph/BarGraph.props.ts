export interface BarGraphProps {
  width: number;
  baseWidth: string;
}
