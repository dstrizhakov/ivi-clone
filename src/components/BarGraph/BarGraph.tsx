import React, { FC } from 'react';
import { BarGraphProps } from '@/components/BarGraph/BarGraph.props';
import styles from './BarGraph.module.scss';

const BarGraph: FC<BarGraphProps> = ({ width }): JSX.Element => {
  if (width > 100) {
    width = 100;
  } else if (width < 0) {
    width = 0;
  }
  return (
    <div className={styles.bar}>
      <div className={styles.barValue} style={{ width: `${width}%` }} data-testid={'bar'} />
    </div>
  );
};

export default BarGraph;
