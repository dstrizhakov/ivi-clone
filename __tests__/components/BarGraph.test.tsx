import BarGraph from '../../src/components/BarGraph/BarGraph';
import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';

describe('BarGraphSnapshot', () => {
  test('25%', () => {
    const bar = render(<BarGraph width={25} />);
    expect(bar).toMatchSnapshot();
  });
  test('50%', () => {
    const bar = render(<BarGraph width={50} />);
    expect(bar).toMatchSnapshot();
  });
  test('75%', () => {
    const bar = render(<BarGraph width={75} />);
    expect(bar).toMatchSnapshot();
  });
  test('100%', () => {
    const bar = render(<BarGraph width={100} />);
    expect(bar).toMatchSnapshot();
  });
});

describe('BarGraph', () => {
  it('should be in the document', async () => {
    render(<BarGraph width={100} />);
    expect(screen.getByTestId('bar')).toBeInTheDocument();
  });

  afterEach(cleanup);
});
