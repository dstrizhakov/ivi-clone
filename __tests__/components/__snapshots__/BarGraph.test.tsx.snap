// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarGraphSnapshot 25% 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bar"
      >
        <div
          class="barValue"
          data-testid="bar"
          style="width: 25%;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bar"
    >
      <div
        class="barValue"
        data-testid="bar"
        style="width: 25%;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BarGraphSnapshot 50% 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bar"
      >
        <div
          class="barValue"
          data-testid="bar"
          style="width: 50%;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bar"
    >
      <div
        class="barValue"
        data-testid="bar"
        style="width: 50%;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BarGraphSnapshot 75% 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bar"
      >
        <div
          class="barValue"
          data-testid="bar"
          style="width: 75%;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bar"
    >
      <div
        class="barValue"
        data-testid="bar"
        style="width: 75%;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BarGraphSnapshot 100% 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bar"
      >
        <div
          class="barValue"
          data-testid="bar"
          style="width: 100%;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bar"
    >
      <div
        class="barValue"
        data-testid="bar"
        style="width: 100%;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
